src/ripple/protocol/impl/STVar.cpp

```diff
@@ -61,37 +59,42 @@ STVar::STVar (STVar&& other)
     }
     else
     {
-        p_ = other.p_->move(
-            sizeof(d_), &d_);
+        p_ = other.p_->move(max_size, &d_);
     }
 }
 
 STVar&
 STVar::operator= (STVar const& rhs)
 {
-    destroy();
-    p_ = nullptr;
-    if (rhs.p_)
-        p_ = rhs.p_->copy(
-            sizeof(d_), &d_);
+    if (&rhs != this)
+    {
+        destroy();
+        if (rhs.p_)
+            p_ = rhs.p_->copy(max_size, &d_);
+        else
+            p_ = nullptr;
+    }
+
     return *this;
 }
 
 STVar&
 STVar::operator= (STVar&& rhs)
 {
-    destroy();
-    if (rhs.on_heap())
+    if (&rhs != this)
```

Comment:
I wanted to avoid accidents along the lines of `x = std::move(x)` since they'd be catastrophic.